src/graphapi/pointwise.cpp

```diff
@@ -246,6 +252,416 @@ void BackendPointwiseDescriptor::getAttribute(miopenBackendAttributeName_t attri
+                       mOperationPointwise.mX->getDimensions().cend(),
+                       mOperationPointwise.mY->getDimensions().cbegin(),
+                       mOperationPointwise.mY->getDimensions().cend()))
+        {
+            MIOPEN_THROW(miopenStatusBadParm);
+        }
+        break;
+
+    /* 3-inputs operations
+     * x input
+     * b input
+     * t input
+     * y output
+     */
+    case MIOPEN_POINTWISE_BINARY_SELECT:
+        if(mOperationPointwise.mX == nullptr || mOperationPointwise.mB == nullptr ||
+           mOperationPointwise.mT == nullptr || mOperationPointwise.mY == nullptr ||
+           mOperationPointwise.mDx != nullptr || mOperationPointwise.mDy != nullptr ||
+           !std::equal(mOperationPointwise.mX->getDimensions().cbegin(),
+                       mOperationPointwise.mX->getDimensions().cend(),
+                       mOperationPointwise.mB->getDimensions().cbegin(),
+                       mOperationPointwise.mB->getDimensions().cend()) ||
+           !std::equal(mOperationPointwise.mX->getDimensions().cbegin(),
+                       mOperationPointwise.mX->getDimensions().cend(),
+                       mOperationPointwise.mT->getDimensions().cbegin(),
+                       mOperationPointwise.mT->getDimensions().cend()) ||
+           !std::equal(mOperationPointwise.mX->getDimensions().cbegin(),
+                       mOperationPointwise.mX->getDimensions().cend(),
+                       mOperationPointwise.mY->getDimensions().cbegin(),
+                       mOperationPointwise.mY->getDimensions().cend()))
+        {
+            MIOPEN_THROW(miopenStatusBadParm);
+        }
+        break;
+
+    /* 2-inputs backward operations
+     * y input
+     * dy input
+     * dx output
+     */
+    case MIOPEN_POINTWISE_RELU_BWD:
+    case MIOPEN_POINTWISE_TANH_BWD:
+    case MIOPEN_POINTWISE_SIGMOID_BWD:
+    case MIOPEN_POINTWISE_ELU_BWD:
+    case MIOPEN_POINTWISE_GELU_BWD:
+    case MIOPEN_POINTWISE_SOFTPLUS_BWD:
+    case MIOPEN_POINTWISE_SWISH_BWD:
+    case MIOPEN_POINTWISE_GELU_APPROX_TANH_BWD:
+        if(mOperationPointwise.mY == nullptr || mOperationPointwise.mDy == nullptr ||
+           mOperationPointwise.mDx == nullptr || mOperationPointwise.mX != nullptr ||
+           mOperationPointwise.mB != nullptr || mOperationPointwise.mT != nullptr ||
+           !checkDimsWithPossibleBroadcasting(mOperationPointwise.mY->getDimensions(),
+                                              mOperationPointwise.mDy->getDimensions(),
+                                              mOperationPointwise.mDx->getDimensions()))
+        {
+            MIOPEN_THROW(miopenStatusBadParm);
+        }
+        break;
+
+    /* TODO: Implement the remaining cases
+     */
+    case MIOPEN_POINTWISE_ERF:
+    case MIOPEN_POINTWISE_GEN_INDEX: MIOPEN_THROW(miopenStatusNotImplemented);
+
+    default: MIOPEN_THROW(miopenStatusNotImplemented);
+    }
```

Comment:
@CAHEK7 @DrizztDoUrden @amberhassaan Colleages, could you check the switch statement, please? Are the pointwise operations grouped correctly, are their inputs and outputs correct, what group should we put the last ones to; and do we support all of the operations or should we throw miopenStatusNotImplemented in certain cases?

Reply:
@scerzh 
> should we throw miopenStatusNotImplemented in certain cases?

If some operation is listed in the API but not implemented, then it seems obvious that we should throw `miopenStatusNotImplemented` instead of `miopenStatusBadParm`. So I recommend exchanging BadParm <-> Not Implemented in this switch.